### PR TITLE
update dips version, fix readme, and set file name as optional param

### PR DESCRIPTION
**Pull request by feixionglee**
update deps version, fix readme, and set file name as optional param for uploader

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,24 @@ config :qiniu, Qiniu,
   secret_key: "secret"
 ```
 
+### Phoemix
```
Comment (tony612) [inline README.md]: typo?

File: mix.exs [hunk 2]
```diff
@@ -1,12 +1,12 @@
 defmodule Qiniu.Mixfile do
   use Mix.Project
 
-  @version "0.2.2"
+  @version "0.2.3"
```
Comment (tony612) [inline mix.exs]: You should not change this. I'll do that when I publish a new verison.

File: README.md [hunk 3]
```diff
@@ -30,8 +48,8 @@ uptoken = Qiniu.Auth.generate_uptoken(policy)
 Upload a local file in server
 
 ```elixir
-put_policy = Qiniu.PutPolicy("books")
-Qiniu.Uploader.upload put_policy, "~/cool.jpg", key: "cool.jpg"
+put_policy = Qiniu.PutPolicy.build("books")
+Qiniu.Uploader.upload put_policy, "/Users/tony612/cool.jpg", "my-cool.jpg"
```
Comment (tony612) [inline README.md]: I don't think we should change this line because you change `:key` to optional.

File: lib/qiniu/uploader.ex [hunk 4]
```diff
@@ -24,25 +24,24 @@ defmodule Qiniu.Uploader do
 
     * `put_policy` - PutPolicy struct
     * `local_file` - path of local file
-    * `key`        - key in a Qiniu bucket
 
   ## Options
-
+    * `key`    - file name in a Qiniu bucket
```
Comment (tony612) [inline lib/qiniu/uploader.ex]: It's better to change `key` to `:key` :)

**Comment by tony612** [comment]
I left some comments :)
Can you squash the two `fix readme` commits and three commits for `mix.exs`(_little_, _fix deps_ and _upgrade poison version_) to keep the commit history clean?

**Comment by feixionglee** [comment]
I have do some fix and clean the commits, please check it, thanks

**Comment by tony612** [comment]
@feixionglee Thank you!

**Pull request merged**